src/runtime/runtime.ts

```diff
@@ -1140,8 +1190,12 @@ export class Transaction {
       index.insert(change);
       changeIx++;
     }
-  }
 
+    // Once the transaction is effectively done, we need to clean up after ourselves. We
+    // arena allocated a bunch of IDs related to function call outputs, which we can now
+    // safely release.
+    GlobalInterner.releaseArena("functionOutput");
```

Comment:
Using a single fixed arena is fine until we go concurrent. We should consider just using some unique info off the transaction to create our arena.